### PR TITLE
convert People / PeopleRow / Person to use hooks

### PR DESCRIPTION
**Pull request by christalee**
I refactored these components together because they share state, in the form of niceties that have been written and updated. I wasn't sure if threading the hooks `updatedNiceties/setUpdatedNiceties` through PeopleRow was the right idiom, but it worked, so I did it.

I "tested" my changes by loading the app, writing and saving some niceties, checking that they appear properly on the Admin page, and remain visible when I switch back to the Write Niceties page.

### CODE REVIEW

File: src/components/People.js [hunk 1]
```diff
@@ -87,77 +81,81 @@ class People extends React.Component {
...
+  let maybeHR = '';
+  if (staying.length > 0) {
+      maybeHeader = (<h3>In-Batch</h3>);
+      maybeHR = (<hr />);
+  }
+  if (justSaved) {
+      alertTimer();
+  }
 
+  const leavingRows = leaving.map((row) => (
```
Comment (christalee) [inline src/components/People.js:108]: I thought it was weird that staffRows was pulled out into its own variable, so I made parallel leavingRows and stayingRows.

File: src/components/Person.js [hunk 2]
```diff
@@ -1,131 +1,127 @@
...
+    const [anonValue, setAnonValue] = useState(false);
+    const [noReadValue, setNoReadValue] = useState(false);
 
-        this.state = {
-            textValue: textValue,
-            anonValue: anonValue,
-            noReadValue: noReadValue,
-        }
-    }
+    const updateState = () => {
```
Comment (christalee) [inline src/components/Person.js:12]: This is a terrible name for a function but I couldn't describe this function better, as it is long and unwieldy.

File: src/components/Person.js [hunk 3]
```diff
@@ -1,131 +1,127 @@
...
+          <div className="person">
+              <Image responsive={true} src={props.data.avatar_url} title={props.data.full_name} circle={true}/>
+              <h3>{props.data.name}</h3>
+              <textarea
+                  defaultValue={textValue}
+                  onChange={textareaChange}
+                  rows="6"
+                  placeholder={props.data.placeholder}/>
+              <Checkbox
+                  checked={anonValue === true}
```
Comment (christalee) [inline src/components/Person.js:114]: Could I simplify these to `checked={anonValue}`?

**Comment by jasonaowen** [comment]
I'm sorry for the delay, @christalee! I drafted these weeks ago and then didn't actually hit submit.

File: src/components/People.js [hunk 4]
```diff
@@ -58,20 +52,20 @@ class People extends React.Component {
         body: JSON.stringify({'niceties': data_to_save}),
       })
       .then(response => response.json())
-      .then(data => {
-        this.setState({noSave: true});
-        this.setState({justSaved: true});
+      .then(() => {
```
[the remark this answers is not in the document]
Comment (christalee) [reply src/components/People.js]: You're right, I failed to actually make the change. Thanks for catching my error.

**Review by jasonaowen — APPROVED** [review]
Thank you so much, @christalee! These look good to me. Yay hooks! :hook:

**Pull request merged**